3023: give Github Actions PR and master branch builds descriptive version names

ericwa opened the pull request:
Adjust GET_GIT_DESCRIBE macro behaviour:

- If "git describe" fails (e.g. because it's a shallow clone, from GitHub Actions building a PR or a commit on `master`), try to use GITHUB_SHA and GITHUB_REF to construct a replacement version string (see: https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables )
- The version string is `{GITHUB_REF}-{GITHUB_SHA}`, with forward slashes replaced with underscores (so the package filename will be valid)
- Behavior should be unchanged for building release tags

Also make it non-fatal if the version description isn't in the v0000.0 format (it won't be for the GitHub Actions master branch / PR cases).

--- 

The Windows build for this PR gets the following package name: 

- `TrenchBroom-Win64-refs_pull_3771_merge-8cee1ee2739547ad2cf21214ff4f3ab1078470b8-Release.7z` 

and the following welcome screen:

![Capture](https://user-images.githubusercontent.com/239161/110064346-281cb300-7d2a-11eb-90b0-2b2c1b4329ba.PNG)

Limitations to note:

- the commit displayed for PR's is the temporary merge commit the GitHub actions makes. I'm not sure if this is possible to trace back to the commit on the PR branch. e.g. the screenshot was taken from the 3rd commit on this branch, 397cc94 . With more work we can probably get the SHA of the commit on the PR branch, but this is probably good enough for now
- The version number is 0.0 for these builds. Since we're working with shallow clones on GitHub Actions, we don't know the most recent tag. This I don't see as a huge problem (it will have the correct version for tagged release builds), although if we care we can switch to non-shallow clones for CI (wasting bandwidth / time though).

Fixes #3023

kduske commented:
We could increase the likelyhood of getting the tag by cloning with a depth of say 100 commits, but I'm actually fine with it being 0000.0 since that makes it obvious that it's an unstable build. Come to think of it, maybe we could show the string "unstable" or "nightly" or something to that effect in this case.

It would however be much better to show the commit hash that's being built. It would make tracing error reports easier. If cloning with depth=100 would fix this, I'd consider it.

ericwa commented:
> Come to think of it, maybe we could show the string "unstable" or "nightly" or something to that effect in this case.

Yeah that sounds good, I could just prepend `unstable-` so it's something like `unstable-refs_pull_3771_merge-8cee1ee2739...`

> It would however be much better to show the commit hash that's being built. It would make tracing error reports easier. 

Right, yeah. I'll check the GitHub actions docs again and see if I can do that

ericwa commented:
Here's the updated look.
- added "unstable-" prefix
- added word wrap to the QLabel, so it all fits on screen
- it uses the commit SHA of the commit on the PR's branch, before the temporary merge commit. We can see it's working here by picking up the right hash, df787c5 
- Builds off of master branch should say something like `unstable-refs_heads_master-<commit SHA on master branch>`. It's working as far as I can test without merging this :)
![Capture](https://user-images.githubusercontent.com/239161/110195492-9f217c80-7dfa-11eb-9711-a97a8aaa9f44.PNG)

kduske reviewed (approved):
Looks great now, thanks for considering my feedback!
